### PR TITLE
Add template protection function

### PR DESCRIPTION
**Pull request by xavierclements**
*Issue #, if available:* None

*Description of changes:*

Add an example bash function to add template protection subject to the environment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -163,13 +185,15 @@ then
         --profile ${DEVOPS_PROFILE}
     echo "Waiting for stack to be created ..."
     aws cloudformation wait stack-create-complete --profile ${DEVOPS_PROFILE} --region ${REGION} --stack-name ${STACK_NAME}
+
+    template_protection "mgmt" $STACK_NAME $REGION $DEVOPS_PROFILE
```
Comment (jaidisido) [inline deploy.sh:189]: Any reason why termination protection should still be enabled on this stack when `ENV` is `dev`?

**Review by jaidisido — COMMENTED** [review]
This would only enable `termination protection` for a fraction of the stacks in the framework. Is your idea to replicate this for other `deploy` scripts across the codebase?

File: deploy.sh (hunk 1 above)
Comment (xavierclements) [reply deploy.sh:189]: @jaidisido fair. I think the reasoning I applied here was that the stack is in the DevOps / Management account so I wanted it protected. I'll change it.

**Comment by xavierclements** [comment]
> This would only enable `termination protection` for a fraction of the stacks in the framework. Is your idea to replicate this for other `deploy` scripts across the codebase?

@jaidisido In the variant that I have it is put in every single `deploy.sh`. Per offline conversation I was posting up patterns and then leaving it to be adopted. I can add it in to the core blocks but might see this PR hang for a bit?

**Pull request merged**